src/main/scala/io/qbeast/spark/delta/IndexStatusBuilder.scala

```diff
@@ -48,37 +53,30 @@ private[delta] class IndexStatusBuilder(
    */
   def buildCubesStatuses: SortedMap[CubeId, CubeStatus] = {
 
-    val spark = SparkSession.active
-    import spark.implicits._
     val rev = revision
     val builder = SortedMap.newBuilder[CubeId, CubeStatus]
     revisionFiles
-      .groupByKey(_.tags(TagUtils.cube))
-      .mapGroups((cube, f) => {
+      .collect()
```

Comment:
why did you change this? Why are we aggregating in-memory in the master instead of using Spark? What was the need for this?

Reply:
Discussion on https://github.com/Qbeast-io/qbeast-spark/pull/65#issuecomment-1022232484 :smile: